examples/src/main/java/com/datastax/oss/driver/examples/mapper/KillrVideoMapperExample.java

```diff
@@ -0,0 +1,192 @@
+      System.out.printf("Videos for %s %s:%n", user.getFirstname(), user.getLastname());
+      for (UserVideo userVideo : userVideos) {
+        System.out.printf("  [%s] %s%n", userVideo.getVideoid(), userVideo.getName());
+      }
+
+      PagingIterable<LatestVideo> latestVideos = videoDao.getLatest(todaysTimestamp());
+      System.out.println("Latest videos:");
+      for (LatestVideo latestVideo : latestVideos) {
+        System.out.printf("  [%s] %s%n", latestVideo.getVideoid(), latestVideo.getName());
+      }
+
+      PagingIterable<VideoByTag> videosByTag = videoDao.getByTag("apachecassandra");
+      System.out.println("Videos tagged with apachecassandra:");
+      for (VideoByTag videoByTag : videosByTag) {
+        System.out.printf("  [%s] %s%n", videoByTag.getVideoid(), videoByTag.getName());
+      }
+
+      // Update the existing video:
+      Video template = new Video();
+      template.setVideoid(video.getVideoid());
+      template.setName(
+          "Getting Started with DataStax Apache Cassandra® as a Service on DataStax Constellation");
+      videoDao.update(template);
+      // Reload the whole entity and check the fields
+      video = videoDao.get(video.getVideoid());
+      System.out.printf("Updated name for video %s: %s%n", video.getVideoid(), video.getName());
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  private static void tryLogin(UserDao userDao, String email, String password) {
+    Optional<User> maybeUser = userDao.login(email, password.toCharArray());
+    System.out.printf(
+        "Logging in with %s/%s: %s%n",
+        email, password, maybeUser.isPresent() ? "Success" : "Failure");
+  }
+
+  public static void maybeCreateSchema(CqlSession session) throws Exception {
+    session.execute(
+        SimpleStatement.newInstance(
+                "CREATE KEYSPACE IF NOT EXISTS killrvideo WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}")
+            .setExecutionProfileName("slow"));
+    session.execute("USE killrvideo");
+    for (String statement : getStatements("killrvideo_schema.cql")) {
+      session.execute(SimpleStatement.newInstance(statement).setExecutionProfileName("slow"));
+    }
+  }
+
+  private static List<String> getStatements(String fileName) throws Exception {
+    Path path = Paths.get(ClassLoader.getSystemResource(fileName).toURI());
+    String contents = new String(Files.readAllBytes(path), StandardCharsets.UTF_8);
+    return Arrays.stream(contents.split(";"))
+        .map(String::trim)
+        .filter(s -> !s.isEmpty())
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * KillrVideo uses a textual timestamp to partition recent video. Build the timestamp for today to
+   * fetch our latest insertions.
+   */
+  private static String todaysTimestamp() {
+    return DateTimeFormatter.ofPattern("yyyyMMdd").withZone(ZoneOffset.UTC).format(Instant.now());
```

Comment:
This is repeated in `CreateVideoQueryProvider`, can't we factor the two together?